sw/host/tests/chip/jtag/src/main.rs

```diff
@@ -51,6 +51,79 @@ fn test_openocd(opts: &Opts, transport: &TransportWrapper) -> Result<()> {
     let result = console.interact(&*uart, None, Some(&mut std::io::stdout()))?;
     log::info!("result: {:?}", result);
 
+    //
+    // Test the RISC-V TAP
+    //
+    reset(
+        transport,
+        &["PINMUX_TAP_RISCV"],
+        opts.init.bootstrap.options.reset_delay,
+    )?;
+
+    let jtag = transport.jtag(&opts.jtag)?;
+    jtag.connect(JtagTap::RiscvTap)?;
+    jtag.halt()?;
+    // Test reads by checking the LC_STATE register
+    //
+    // TODO use bindgen to get the TOP_EARLGREY_LC_CTRL_BASE_ADDR and
```

Comment:
could you use this autogenerated file: https://cs.opensource.google/opentitan/opentitan/+/master:hw/top_earlgrey/sw/autogen/chip/top_earlgrey.rs

(same below for the retention SRAM base addr)

Reply:
Done. I couldn't figure out how to just add the sources to the test because it's difficult to use a module that is at a different place in the file hierarchy, I ended create a rust library for them.